deepchem/models/torch_models/ferminet.py

```diff
@@ -169,6 +174,86 @@ def loss(self,
 
+    def calculate_nuclear_nuclear(self,) -> torch.Tensor:
+        """
+        Function to calculate where only the nucleus terms are involved and does not change when new electrons are sampled.
+        atom-atom potential term = Zi*Zj/|Ri-Rj|, where Zi, Zj are the nuclear charges and Ri, Rj are nuclear coordinates
+
+        Returns:
+        --------
+        A torch tensor of a scalar value containing the nuclear-nuclear potential term (does not change for the molecule system with sampling of electrons)
+        """
+
+        potential = torch.nan_to_num(
+            (self.nuclear_charge * 1 /
+             torch.cdist(self.nucleon_pos.float(), self.nucleon_pos.float()) *
+             self.nuclear_charge.unsqueeze(1)),
+            posinf=0.0,
+            neginf=0.0)
+        potential = torch.sum(potential) / 2
+        return potential
+
+    def calculate_electron_nuclear(self,) -> torch.Tensor:
+        """
+        Function to calculate the expected electron-nuclear potential term per batch
+        nuclear-electron potential term = Zi/|Ri-rj|, rj is the electron coordinates, Ri is the nuclear coordinates, Zi is the nuclear charge
+
+        Returns:
+        --------
+        A torch tensor of a scalar value containing the electron-nuclear potential term.
+        """
+
+        potential = torch.sum(
+            (1 / torch.cdist(self.input.float(), self.nucleon_pos.float())) *
+            self.nuclear_charge) / 2
+        return (potential / self.batch_size)
+
+    def calculate_electron_electron(self,):
+        """
+        Function to calculate the expected electron-nuclear potential term per batch
+        nuclear-electron potential term = 1/|ri-rj|, ri, rj is the electron coordinates
+
+        Returns:
+        --------
+        A torch tensor of a scalar value containing the electron-electron potential term.
+        """
+        potential = torch.sum(
+            torch.nan_to_num(
+                (1 / torch.cdist(self.input.float(), self.input.float())),
+                posinf=0.0,
+                neginf=0.0)) / 2
+        return (potential / self.batch_size)
+
+    def calculate_kinetic_energy(self,):
+        """
+        Function to calculate the expected kinetic energy term per batch
+        It is calculated via:
+        \sum_{ri}^{}[(\pdv[]{log|\Psi|}{(ri)})^2 + \pdv[2]{log|\Psi|}{(ri)}]
+
+        Returns:
+        --------
+        A torch tensor of a scalar value containing the electron-electron potential term.
+        """
+        log_probability = torch.log(torch.abs(self.psi))
+        jacobian = list(
+            map(
+                lambda x: torch.autograd.grad(x, self.input, create_graph=True)[
+                    0], log_probability))
+        jacobian_square = list(
+            map(lambda x: torch.sum(torch.pow(x, 2)), jacobian))
+        jacobian_square_sum = torch.tensor(0.0)
+        hessian = torch.tensor(0.0)
+        for i in range(self.batch_size):
```

Comment:
We may want to use https://pytorch.org/docs/master/generated/torch.vmap.html#torch.vmap instead to speed this up. Can be done in future PR